### PR TITLE
New capabilities: mesh sampling, voxelization, and project conversion

### PR DESCRIPTION
**Pull request by dhuber23b**

### CODE REVIEW

File: sumo-api/sumo/threedee/model_sampler.cpp [hunk 1]
```diff
@@ -0,0 +1,7268 @@
+/* Generated by Cython 0.27.3 */
```
Comment (NCBee) [inline sumo-api/sumo/threedee/model_sampler.cpp]: This file should not be in the diff

File: sumo-api/sumo/threedee/model_sampler.pyx [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        counts = [m.num_indices() for m in model.primitive_meshes()]
+        n_faces3 = np.sum(counts)  # 3 x num faces
+        faces = np.zeros((n_faces3, 6))
+
+        # build faces list from element component meshes
+        start = 0
+
+        for i, mesh in enumerate(model.primitive_meshes()):
+            # replicate verts based on face ids
+            # temp[:,i] = vertex for face_id i
```
Comment (NCBee) [inline sumo-api/sumo/threedee/model_sampler.pyx]: Remove

File: sumo-api/sumo/threedee/tests/model_sampler_tests.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        """
+        input_path = os.path.join(TEST_PATH, "bed.glb")
+        model = GltfModel.load_from_glb(input_path)
+        sampler = ModelSampler()
+        samples = sampler.run(model)
+
+        self.assertEqual(samples.shape[1], 6)
+        self.assertTrue(samples.shape[0] > 0)
+
+# This code will manually visualize the samples to verify that it is working.
```
Comment (NCBee) [inline sumo-api/sumo/threedee/tests/model_sampler_tests.py]: Remove block

File: sumo-api/sumo/threedee/tests/voxelizer_tests.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        input_path = os.path.join(TEST_PATH, "bed.glb")
+        model = GltfModel.load_from_glb(input_path)
+        voxelizer = Voxelizer()
+        voxels = voxelizer.run(model)
+        centers = voxels.voxel_centers()
+        
+        self.assertEqual(centers.shape[1], 3)
+        self.assertTrue(centers.shape[0] > 0)
+
+# This code will manually visualize the samples to verify that it is working.
```
Comment (NCBee) [inline sumo-api/sumo/threedee/tests/voxelizer_tests.py]: Remove block

File: sumo-api/sumo/threedee/model_sampler.cpp (hunk 1 above)
Comment (dhuber23b) [reply sumo-api/sumo/threedee/model_sampler.cpp]: oops.

File: sumo-api/sumo/threedee/model_sampler.pyx (hunk 2 above)
Comment (dhuber23b) [reply sumo-api/sumo/threedee/model_sampler.pyx]: this is not code.  It is a comment to explain the structure

File: sumo-api/sumo/threedee/tests/voxelizer_tests.py (hunk 4 above)
Comment (dhuber23b) [reply sumo-api/sumo/threedee/tests/voxelizer_tests.py]: This commented code serves as a manual test.  No easy to automate.

**Pull request merged**